### DISCUSSION TITLE
Cling and 11.1 SDK

### DISCUSSION THREAD

**Discussion by spinicist** [Homebrew/discussions#504] [Casks]
Hello,

I'm not sure if this is the right place to ask for help. I installed the Cling interpreter from homebrew today. Sadly, when starting I get the following message:

```
warning: no such sysroot directory: '/Library/Developer/CommandLineTools/SDKs/MacOSX11.0.sdk'
In file included from input_line_1:1:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/new:90:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/exception:81:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/cstdlib:85:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/stdlib.h:97:15: fatal error: 'stdlib.h' file not found
#include_next <stdlib.h>
```

And indeed, I do not have the 11.0 SDK installed, because I have the 11.1 SDK installed. Is this an error in Cling or in the recipe?

(I also posted to the Cling forum here: https://root-forum.cern.ch/t/homebrew-and-latest-macos-sdk/43049)

Thanks in advance for any help.

**Comment by carlocab** [comment]
What is the output of the following commands?
```
brew info cling
brew config
brew doctor
```

**Comment by spinicist** [comment]
Hello,

```
➜  t1w brew info cling
cling: stable 0.7 (bottled)
C++ interpreter
https://root.cern.ch/cling
/usr/local/Cellar/cling/0.7 (2,065 files, 951.2MB) *
  Poured from bottle on 2021-01-11 at 08:57:34
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/cling.rb
License: LGPL-2.1-only or NCSA
==> Dependencies
Build: cmake ✔
==> Analytics
install: 338 (30 days), 518 (90 days), 1,247 (365 days)
install-on-request: 337 (30 days), 515 (90 days), 1,243 (365 days)
build-error: 0 (30 days)
```

```
➜  t1w brew config
HOMEBREW_VERSION: 2.7.2
ORIGIN: https://github.com/Homebrew/brew
HEAD: dad7dc6a1498b80770d98f2d7cd6fb927c300bbb
Last commit: 5 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 745fcd5a474c253e5bc05813b4d777de43cb0ed3
Core tap last commit: 8 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.Oit4cfQZul/org.macosforge.xquartz:0
HOMEBREW_MAKE_JOBS: 6
Homebrew Ruby: 2.6.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby
CPU: hexa-core 64-bit kabylake
Clang: 12.0 build 1200
Git: 2.28.0 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.1-x86_64
CLT: 12.3.0.0.1.1607026830
Xcode: 12.3
XQuartz: 2.7.11 => /opt/X11
```

```
➜  t1w brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Homebrew's sbin was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting the PATH for example like so:
  echo 'export PATH="/usr/local/sbin:$PATH"' >> ~/.zshrc
```

**Reply by carlocab** [reply]
Hm, yes, I'm able to reproduce this problem as well. However, it's not clear to me what the cause is.

Can you try using a non-Homebrew version of Cling, and see if the error is not present there?

**Reply by spinicist** [reply]
Hello,

I downloaded the Cling binary snapshot from https://root.cern/download/cling/  from 5th November and it starts without complaint. It looks like that was built on 10.15 though, and the snapshot includes a full LLVM toolchain (i.e. there is a copy of clang in there).

I'm a complete newbie to Cling, so I don't know what's expected to work. It looks to me that if I followed the build from source instructions here: https://root.cern/cling/cling_build_instructions/ that it would also build a full LLVM toolchain. But the Homebrew version error messages suggests it is trying to use the system compiler?

**Comment by spinicist** [comment]
One of the Cling developers has opened an issue to track this: https://github.com/root-project/root/issues/7021

**Reply by Axel-Naumann** [reply]
Yeah that's certainly on us (https://github.com/root-project/root/issues/7021), not sure how Homebrew could fix this on their side.

**Comment by razzius** [comment]
This isn't an issue in v0.9 which is [now on homebrew](https://github.com/Homebrew/homebrew-core/commit/7584c413978c2a127c5e79c64549ff66484a6bf7)! Try upgrading
